file/join.rs

```diff
@@ -0,0 +1,224 @@
+            for num in o.clone().unwrap() {
+                let f_num: Vec<&str> = num.split('.').collect();
+                if f_num[0] == "1" {
+                    res.push(s[f_num[1].parse::<usize>().unwrap() - 1].clone());
+                }
+                if f_num[0] == "2" {
+                    if i >= file2.len() && e.is_some() {
+                        res.push(e.clone().unwrap());
+                    }
+                    else {
+                        res.push(file2[i][f_num[1].parse::<usize>().unwrap() - 1].clone());
+                    }
+                }
+            }
+            i += 1;
+            
+            println!("{}", res.join(" "));
+            if i >= file2.len() && e.is_none() {
+                break;
+            }
+        }
+        return Ok(());
+    }
+
+    let mut map: HashMap<String, Vec<Vec<String>>> = HashMap::new();
+
+    for line in &file1 {
+        let key = line[field1 - 1].clone();
+        map.entry(key).or_insert_with(Vec::new).push(line.clone());
+    }
+
+    let mut matched: HashMap<String, bool> = HashMap::new();
+
+    for line in &file2 {
+        let key = line[field2 - 1].clone();
+        if let Some(matches) = map.get_mut(&key) {
+            matched.insert(key.clone(), true);
+            for l in matches.iter_mut() {
+                let mut output = vec![];
+
+                output.extend_from_slice(&l);
+                output.extend_from_slice(&line[1..]);
+
+                if v.unwrap_or(0) != 1 && v.unwrap_or(0) != 2 {
+                    println!("{}", output.join(" "));
+                }
+            }
+        } else if a.unwrap_or(0) == 2 {
+            let mut output = vec![String::from("(unknown)")];
+            output.extend_from_slice(&line[1..]);
+            println!("{}", output.join(" "));
+        }
+    }
+
+    let mut map1: HashMap<String, Vec<String>> = HashMap::new();
+    let mut map2: HashMap<String, Vec<String>> = HashMap::new();
+
+    for line in &file1 {
+        let key = line[field1 - 1].clone();
+        map1.insert(key, line.clone());
+    }
+
+    for line in &file2 {
+        let key = line[field2 - 1].clone();
+        map2.insert(key, line.clone());
+    }
+
+    if v.unwrap_or(0) == 1 {
+        for (key, line1) in &map1 {
+            if !map2.contains_key(key) {
+                println!("{}", line1.join(" "));
+            }
+        }
+    }
+
+    if v.unwrap_or(0) == 2 {
+        for (key, line1) in &map2 {
+            if !map1.contains_key(key) {
+                println!("{}", line1.join(" "));
+            }
+        }
+    }
+
+    if a.unwrap_or(0) == 1 {
+        for (key, line1) in &map1 {
+            if !map2.contains_key(key) {
+                println!("{}", line1.join(" "));
+            }
+        }
+    }
+
+    if a.unwrap_or(0) == 2 {
+        for (key, line1) in &map2 {
+            if !map1.contains_key(key) {
+                println!("{}", line1.join(" "));
+            }
+        }
+    }
```

Comment:
Just set default values for those args, so you don't have to unwrap with default every time.